build/source/dshare/outpt_stat.f90

```diff
@@ -0,0 +1,294 @@
+! SUMMA - Structure for Unifying Multiple Modeling Alternatives
+! Copyright (C) 2014-2015 NCAR/RAL
+!
+! This file is part of SUMMA
+!
+! For more information see: http://www.ral.ucar.edu/projects/summa
+!
+! This program is free software: you can redistribute it and/or modify
+! it under the terms of the GNU General Public License as published by
+! the Free Software Foundation, either version 3 of the License, or
+! (at your option) any later version.
+!
+! This program is distributed in the hope that it will be useful,
+! but WITHOUT ANY WARRANTY; without even the implied warranty of
+! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+! GNU General Public License for more details.
+!
+! You should have received a copy of the GNU General Public License
+! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+! used to manage output statistics of the model and forcing variables
+module output_stats
+USE nrtype
+implicit none
+private
+public :: allocStat
+public :: calcStats
+!public :: compileBasinStats
+contains
+
+ ! ******************************************************************************************************
+ ! public subroutine allocStat is called at beginning of simulation to allocate space for output statistics 
+ ! ******************************************************************************************************
+ subroutine allocStat(meta,stat,err,message)
```

Comment:
Can this be merged with `allocGlobal`?

It would be *wonderful* if this is possible.